File: src/main/java/br/com/jiratorio/domain/entity/UserConfig.java
package br.com.jiratorio.domain.entity;

import br.com.jiratorio.domain.ChartType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity
@EqualsAndHashCode(of = "username", callSuper = false)
public class UserConfig extends BaseEntity {

    private static final long serialVersionUID = -9168105728096346993L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String username;

    private String state;

    private String city;

    private String holidayToken;

    @Enumerated(EnumType.STRING)
    private ChartType leadTimeChartType;

    @Enumerated(EnumType.STRING)
    private ChartType throughputChartType;

}

File: src/main/java/br/com/jiratorio/domain/entity/UserConfig.kt
package br.com.jiratorio.domain.entity

import br.com.jiratorio.domain.chart.ChartType
import br.com.jiratorio.extension.equalsBuilder
import br.com.jiratorio.extension.toStringBuilder
import java.util.Objects
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class UserConfig(

    @Column(nullable = false)
    val username: String,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    @Column(unique = true)
    var state: String? = null,

    var city: String? = null,

    var holidayToken: String? = null,

    @Enumerated(EnumType.STRING)
    var leadTimeChartType: ChartType? = null,

    @Enumerated(EnumType.STRING)
    var throughputChartType: ChartType? = null

) : BaseEntity() {
    companion object {
        private val serialVersionUID = -9168105728096346993L
    }

    override fun toString() =
        toStringBuilder(
            UserConfig::id,
            UserConfig::username,
            UserConfig::state,
            UserConfig::city,
            UserConfig::holidayToken,
            UserConfig::leadTimeChartType,
            UserConfig::throughputChartType
        )

    override fun equals(other: Any?) =
        equalsBuilder(other, UserConfig::username)

    override fun hashCode() =
        Objects.hash(username)

}
